src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtEmailService.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtEmail;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.Email;
+import uk.gov.hmcts.dts.fact.model.admin.EmailType;
+import uk.gov.hmcts.dts.fact.repositories.CourtEmailRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.repositories.EmailTypeRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+@Service
+public class AdminCourtEmailService {
+
+    private final CourtRepository courtRepository;
+    private final CourtEmailRepository emailRepository;
+    private final EmailTypeRepository emailTypeRepository;
+
+    @Autowired
+    public AdminCourtEmailService(final CourtRepository courtRepository,
+                                  final CourtEmailRepository emailRepository,
+                                  final EmailTypeRepository emailTypeRepository) {
+        this.courtRepository = courtRepository;
+        this.emailRepository = emailRepository;
+        this.emailTypeRepository = emailTypeRepository;
+    }
+
+    public List<Email> getCourtEmailsBySlug(final String slug) {
+        return courtRepository.findBySlug(slug)
+            .map(c -> c.getCourtEmails()
+                .stream()
+                .map(CourtEmail::getEmail)
+                .map(Email::new)
+                .collect(toList()))
+            .orElseThrow(() -> new NotFoundException(slug));
+    }
+
+    @Transactional(rollbackFor = {RuntimeException.class})
```

Comment:
I was reading more about this 'transactional' annotation. By default, it will do a rollback for RuntimeException and Error (Not checked exception). May be we should remove the 'rollbackFor' parameter within the annotation and just leave it with the default

Reply:
Changed to use default like you mentioned